cpp/src/arrow/compute/light_array.cc

```diff
@@ -395,8 +395,12 @@ int ExecBatchBuilder::NumRowsToSkip(const std::shared_ptr<ArrayData>& column,
       --num_rows_left;
       int row_id_removed = row_ids[num_rows_left];
       const uint32_t* offsets =
-          reinterpret_cast<const uint32_t*>(column->buffers[1]->data());
+          reinterpret_cast<const uint32_t*>(column->buffers[1]->data()) + column->offset;
```

Comment:
You should probably be able to write:
```c++
      const uint32_t* offsets = column->GetValues<uint32_t>(1);
```

Reply:
Also, I suppose this doesn't work on large binary arrays (where the offset type would be `int64_t`)?

Reply:
You are right, I don't think they work for large binary or large string.

Reply:
Thanks for the simplified code suggestion. That's very nice to do.

Reply:
Addressed.